### PR TITLE
Peregrine Balas

### PR DESCRIPTION
**Pull request by PeregrineReed**

### CODE REVIEW

**Comment by PeregrineReed** [comment]
@iandouglas @memcmahon I'd love to get feedback on this. I hit a roadblock early on with my database not accepting the astronauts I was creating, I don't recall the error off the top of my head, and I kinda buried it in another error. But it wasn't something I knew how to deal with.

File: app/views/astronauts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<h1>Apollo 14</h1>
+<nav>
+  Average Age: <%= @astronauts.average(:age) %>
```
Comment (iandouglas) [inline app/views/astronauts/index.html.erb]: We don't want our view asking the database to calculate things, so this is better put in a model method, that method called by our controller using something like `@avg_age` as an instance variable, and passed to the view

File: app/views/astronauts/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+</nav>
+<div class=<%= "astro-#{astronaut.id}" %>
+<% @astronauts.each do |astronaut| %>
```
Comment (iandouglas) [inline app/views/astronauts/index.html.erb]: lines 5 and 6 should be switched here, you can't put a class or id on the div for that astronaut on line 5 until you have that block parameter available to you on line 6

better syntax for the div as well would be:
`div id="astro-<%= astronaut.id %>">`
ERB is effectively doing one big massive string interpolation for us, so having it do extra interpolation inside the ERB tag can be avoided.

File: app/views/astronauts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+<h4><%= astronaut.name %></h4>
+<p><%= astronaut.age %></p>
```
Comment (iandouglas) [inline app/views/astronauts/index.html.erb]: As I've mentioned in class, it's better to have text content around numbers on a page for testing purposes

File: app/views/astronauts/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<p><%= astronaut.job %></p>
+<ul>
+  <% astronaut.missions.each do |mission| %>
+  <li><%= mission.title %></li>
+  <li><%= mission.time_in_space %></li>
```
Comment (iandouglas) [inline app/views/astronauts/index.html.erb]: None of the stories asked for the time in space to be written on the page. It's really important, especially on timed assessments, to only build what we specifically ask for and nothing else.

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,7 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+@neil = Astronaut.create(name: 'Neil Armstrong', age: 37, job: 'Commander')
+@buzz = Astronaut.create(name: 'Buzz Aldrin', age: 30, job: 'Alien Expert')
+@cosmo = Astronaut.create(name: 'Cosmo Kramer', age: 35, job: 'Comedian')
+@astronauts = [@neil, @buzz, @cosmo]
```
Comment (iandouglas) [inline db/seeds.rb]: this is good, but we only need a seeds file for development mode; these friday challenges can just be done using testing only

File: spec/features/astronauts_index_page_spec.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  it 'sees a list fo astronauts' do
+
+    visit astronauts_path
+
+    @astronauts.each do |astronaut|
+      within "astro-#{astronaut.id}" do
+        expect(page).to have_content(astronaut.name)
+        expect(page).to have_content(astronaut.age)
+        expect(page).to have_content(astronaut.job)
+      end
```
Comment (iandouglas) [inline spec/features/astronauts_index_page_spec.rb]: this is a good test; one extra thing would be testing for content around the numeric age as mentioned before

File: spec/features/astronauts_index_page_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    end
+  end
+
+  it 'sees an average age of all astronauts' do
+
+    visit astronauts_path
+
+    expect(page).to have_content("Average Age: #{Astronaut.average(:age)}")
```
Comment (iandouglas) [inline spec/features/astronauts_index_page_spec.rb]: it's okay to call this AR call in the test itself

**Pull request closed without merging**